src/Microsoft.AspNet.Mvc.Core/Filters/DefaultFilterProvider.cs

```diff
@@ -31,6 +34,10 @@ public void OnProvidersExecuting([NotNull] FilterProviderContext context)
                 {
                     ProvideFilter(context, item);
                 }
+                context.Results =
+                    context.Results
+                    .OrderBy(filter => filter.Descriptor, FilterDescriptorCorsComparer.Comparer)
```

Comment:
I will repeat my self- It is easier to messup the order using Order property than to write a new custom filter provider. 
I really do not see a scenario where a user would want to mess with this. 
In general I do agree that this is more coupled ( and has the downsides that you mentioned). Lets talk to @lodejard and see if he has an opinion on this.